### PR TITLE
feat: Socket authenticators and data transformers

### PR DESCRIPTION
**Pull request by gkc**
**- What I did**
- Added support for requiring client sockets to be authenticated in some 
  app-defined way before they will be connected to the other side
- Added support for app-defined data transformers which can be used to 
  transform the data while sending from A to B, and vice versa. Useful for
  adding traffic encryption, for example.
- Refactored for readability
- Multiple breaking changes to improve API readability
- More documentation
- More tests
- Added GitHub action to run the unit tests

**- How I did it**
- See commits

**- How to verify it**
- Unit test job passes

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,11 @@
+## 1.1.0
```
Comment (xavierchanth) [inline CHANGELOG.md]: Aren't we looking at a major version change? The SocketConnector class had breaking changes

Comment (gkc) [reply CHANGELOG.md]: Updated version and CHANGELOG for 2.0.0 in [this commit](https://github.com/atsign-foundation/socket_connector/pull/4/commits/acc28c31e9ce028dcb16948f2b3ce7c9fcec3b78)

**Review by xavierchanth — APPROVED** [review]
LGTM, my only additional feedback is that we think about disabling the logTraffic flag, and only enabling it with a debug flag, like with our debug build for sshrvd, however maybe not, because this is a library.

**Comment by gkc** [comment]
@XavierChanth @VJag @cconstab I think this is good to go, please review again - main thing to look at since last time is the additions to the test pack. Still marked as draft because I still need to update the main README before the new version is published

**Review by cconstab — APPROVED** [review]
Yay farside

**Pull request merged**